mlflow/utils/databricks_utils.py

```diff
@@ -664,20 +664,29 @@ def _fail_malformed_databricks_auth(uri):
     else:
         uri_name = "tracking URI"
         uri_scheme = "databricks"
+    if is_in_databricks_model_serving_environment():
+        raise MlflowException(
+            f"Reading Databricks credential configuration in model serving failed. "
+            f"Most commonly, this happens because the model currently "
```

Comment:
@smurching   Do Databricks resource dependencies work outside `agents.deploy()` (in other words, does this universally apply to all model serving users)? If not, can we rephrase it so sth like 
```
- If you are using `agents.deploy()`, the model currently being served might have been logged without Databricks resources dependencies .... 
- Otherwise, configure authentication by setting environment variables ....
```

Reply:
> Do Databricks resource dependencies work outside agents.deploy() (in other words, does this universally apply to all model serving users)?

Good question, they do apply to all model serving use cases